nodejs/awsx/apigateway/authorizer.ts

```diff
@@ -0,0 +1,114 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as awslambda from "aws-lambda";
+
+export type AuthorizerEvent = awslambda.CustomAuthorizerEvent;
+export type AuthorizerResponse = awslambda.CustomAuthorizerResult;
+
+/**
+ * SecurityDefinition provides the definition for a custom Authorizer for API Gateway.
+ */
+export interface SecurityDefinition {
+    /**
+     * Pretty name for the security definition to be referenced as. This is only used within the
+     * swagger and must be unique for each unique authorizer within the API. If you want
+     * to share an authorizer across routes, provide the same name for them. If not defined, a unique
+     * name for you.
+     */
+    authorizerName?: string;
+
+    /**
+     * parameterName is the name of the header or query parameter containing the authorization token. Must be
+     * "Unused" for multiple identity sources or non header or query type of request parameters.
+     * */
+    parameterName: string;
+
+    /**
+     * Required and the value must be "header" or "query" for an API Gateway API. Must be "header" for multiple
+     * identity sources or non header or query type of request parameters.
+     */
+    parameterLocation: "header" | "query";
+
+    /**
+     * Specifies the authorization mechanism for the client. Typical values are "oauth2" or
+     * "custom".
+     */
+    authType: string;
+
+    /**
+     * Defines a Lambda authorizer to be applied for authorization of method invocations in API Gateway.
+     */
+    authorizer: LambdaAuthorizer;
+}
+
+/**
+ * Defines a Lambda authorizer (the x-amazon-apigateway-authorizer object) to be applied for authorization of
+ * method invocations in API Gateway.
+ */
+export interface LambdaAuthorizer {
+    /**
+     * The type of the authorizer. This is a required property and the value must be one of the following:
+     *      - "token", for an authorizer with the caller identity embedded in an authorization token
+     *      - "request", for an authorizer with the caller identity contained in request parameters
+     */
+    type: "token" | "request";
+
+    /**
+     * The authorizer specifies information about the authorizing Lambda. You can either set up the Lambda
+     * separately and just provide the required information or you can define the Lambda inline.
+     */
+    authorizer: LambdaInfo | aws.lambda.EventHandler<AuthorizerEvent, AuthorizerResponse>;
+
+    /**
+     * Comma-separated list of mapping expressions of the request parameters as the identity source. Applicable
+     * for the authorizer of the "request" type only.
+     * Example: "method.request.header.HeaderAuth1, method.request.querystring.QueryString1"
+     */
+    identitySource?: string;
```

Comment:
consider taking this as a `string[]` and producing the comma-separated string ourself.